File: Best Time to Buy and Sell Stock II.py
# time complexity is o(n), where n is the size of the input
# space complexity is o(1)
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        res = 0
        for i in range(len(prices)-1):
            if(prices[i] < prices[i+1]):
                res += prices[i+1] - prices[i]
        return res
       

File: Best Time to Buy and Sell Stock II.cpp
//Time Complexity- O(n)
//Space Complexity- O(1)

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int ans=0;
        int minVal=prices[0];
        for(int i=1;i<prices.size();i++){
            if(prices[i]<=minVal){
                minVal=prices[i];
            }
            else{
                ans+=prices[i]-minVal;
                minVal=prices[i];
            }
        }
        return ans;
    }
};